src/frontEnd/ProjectExplorer.py

```diff
@@ -226,51 +230,101 @@ def refreshProject(self):
+        self.baseFileName = str(self.indexItem.data())
+        self.newBaseFileName, ok = QtGui.QInputDialog.getText(self, 'Rename Project', 'Project Name:',
+                                                            QtGui.QLineEdit.Normal, self.baseFileName)
+        if ok and self.newBaseFileName:
+            print("=================")
+            print(self.newBaseFileName)
+            print("=================")
+            self.newBaseFileName = str(self.newBaseFileName)
+            projectPath, projectFiles = list(self.obj_appconfig.project_explorer.items())[self.indexItem.row()]
             updatedProjectFiles = []
 
-            # rename files matching project name
-            for projectFile in projectFiles:
-                if baseFileName in projectFile:
-                    oldFilePath = os.path.join(projectPath, projectFile)
-                    projectFile = projectFile.replace(
-                        baseFileName, newBaseFileName, 1)
-                    newFilePath = os.path.join(projectPath, projectFile)
-                    print ("Renaming " + oldFilePath + " to " + newFilePath)
-                    os.rename(oldFilePath, newFilePath)
-
-                    updatedProjectFiles.append(projectFile)
-
-            # rename project folder
-            updatedProjectPath = newBaseFileName.join(
-                projectPath.rsplit(baseFileName, 1))
-            print ("Renaming " + projectPath + " to " + updatedProjectPath)
-            os.rename(projectPath, updatedProjectPath)
-
-            # update project_explorer dictionary
-            del self.obj_appconfig.project_explorer[projectPath]
-            self.obj_appconfig.project_explorer[updatedProjectPath] = (
-                updatedProjectFiles
-            )
+            self.workspace = self.obj_appconfig.default_workspace['workspace']
+            self.newBaseFileName = str(self.newBaseFileName).rstrip().lstrip()
+            self.projDir = os.path.join(self.workspace, str(self.newBaseFileName))
 
-            # save project_explorer dictionary on disk
-            json.dump(self.obj_appconfig.project_explorer,
-                      open(self.obj_appconfig.dictPath, 'w'))
+            if self.newBaseFileName == "":
+                print("Project name can not be empty")
+                print("==================")
+                msg = QtGui.QErrorMessage(self)
+                msg.showMessage('The project name cannot be empty')
+                msg.setWindowTitle("Error Message")
+
+            elif self.baseFileName == self.newBaseFileName:
+                print("Project name has to be different")
+                print("==================")
+                msg = QtGui.QErrorMessage(self)
+                msg.showMessage('The project name has to be different')
+                msg.setWindowTitle("Error Message")
+
+            else:
+                self.reply = self.obj_validation.validateNewproj(str(self.projDir))
```

Comment:
```suggestion
                reply = self.obj_validation.validateNewproj(str(self.projDir))
```
don't use self. for local variables